community/kernel/src/main/java/org/neo4j/kernel/impl/api/KernelStatement.java

```diff
@@ -107,7 +110,7 @@ public KernelStatement( KernelTransactionImplementation transaction,
         this.facade = new OperationsFacade( transaction, this, procedures );
         this.executingQueryList = ExecutingQueryList.EMPTY;
         this.systemLockTracer = systemLockTracer;
-        this.statementOpenCloseCalls = TRACK_STATEMENTS ? new ArrayDeque<>() : null;
+        this.statementOpenCloseCalls = TRACK_STATEMENTS ? new ArrayDeque<>() : EMPTY_STATEMENT_HISTORY;
```

Comment:
Shouldn't this be `RECORD_STATEMENTS_TRACES` as conditional?
I think returning null would be clearer. See comment below.